db.go

```diff
@@ -852,6 +852,53 @@ func (db *DB) CleanTombstones() (err error) {
 	return errors.Wrap(db.reload(), "reload blocks")
 }
 
+// LabelNames returns all the unique label names present in the DB in sorted order.
+func (db *DB) LabelNames() (labelNames []string, err error) {
+	headIndexr, _ := db.head.Index()
+	defer headIndexr.Close()
+	labelNamesMap := make(map[string]struct{})
+
+	names, err := headIndexr.LabelNames()
+	if err != nil {
+		return nil, err
+	}
+	for _, name := range names {
```

Comment:
Could you close `headIndexr` here itself, as it's not being used again. Further, it returns an error which should be returned.

Also, could you wrap the errors with appropriate messages like: `errors.Wrap(err, "close headIndexReader")`